### ISSUE TITLE
Default systemd unit file and freshclam config not conducive to reporting errors to system admin

### ISSUE THREAD

**Issue by nomis** [Cisco-Talos/clamav#269]
Describe the bug
----------------

The example systemd unit file directs all of the output of freshclam to the journal (syslog) where it will only have log level "info".

Without looking for clamav's specific logging legend characters it is not possible to know the log level:
```
 *  ! - ERROR:
 *  ^ - WARNING:
 *  ~ - normal
 *  # - normal, not foreground (logfile and syslog only)
 *  * - verbose
 *  $ - debug
 *  none - normal
```

Messages at log level "error" or above are automatically emailed to me but updates failed to download for over a month because all of the error messages are at log level "info".

There is another problem here in that unnecessary errors are being deliberately introduced:
```
!cdiff_apply: lseek(desc, -350, SEEK_END) failed
!downloadPatch: Can't apply patch
```

The "empty file" scenario needs to be downgraded to a warning to avoid spurious errors being reported.

How to reproduce the problem
----------------------------

Use a slightly outdated ClamAV with default systemd unit file and logging. Set freshclam up to need updates but fail to download them.

None of the warning/error messages in syslog have a higher log level than "info" because they come via the systemd journal.

The log messages need to go into syslog with the right log level by default so that any monitoring can pick up warning/error messages.

**Comment by micahsnyder** [comment]
> ```
> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
> !downloadPatch: Can't apply patch
> ```
This is a bug that we need to fix in 0.103.4+.  It shouldn't have tried to apply the empty patch.

The printing of the special logging characters (`!^~#*$`) is also a long standing bug that occurs when `LogTime` is enabled. See: https://github.com/Cisco-Talos/clamav/pull/157

But even if we fix the special-character issue, it sounds like there's still may be a bug here per your last comment: "The log messages need to go into syslog with the right log level by default so that any monitoring can pick up warning/error messages."

**Comment by nomis** [comment]
I started matching the pattern of the time and special logging characters but now I've disabled `LogTime` so I can just match on `ERROR:*`.

The configuration files are a complication because they apply to both interactive use (where stdout and times would be preferred) and daemon use (where syslog without times would be preferred). A command line option to override `LogSyslog` and `LogTime` would fix this but I expect people will then complain that messages no longer go into the systemd journal.

It looks like using Linux kernel format priority levels in the output will set the log level for the journal messages: http://0pointer.de/blog/projects/journal-submit.html. That could be a command line parameter that should also disable `LogTime`.

Presumably I should be using `OnErrorExecute` instead, and this could default to emailing `root`?
It's only because my IP got blocked that I noticed a problem because the service had stopped running.

**Comment by micahsnyder** [comment]
I believe this was fixed in 0.105.0, thanks to @mko-x's work ( https://github.com/Cisco-Talos/clamav/pull/445 ).

The logic here is now:
```c
#if defined(USE_SYSLOG) && !defined(C_AIX)
    if (logg_syslog) {
        cli_chomp(buff);
        if (loglevel == LOGG_ERROR) {
            syslog(LOG_ERR, "%s", buff);
        } else if (loglevel == LOGG_WARNING) {
            if (!logg_nowarn)
                syslog(LOG_WARNING, "%s", buff);
        } else if (loglevel == LOGG_DEBUG || loglevel == LOGG_DEBUG_NV) {
            syslog(LOG_DEBUG, "%s", buff);
        } else
            syslog(LOG_INFO, "%s", buff);
    }
#endif
```
